lib/charms/mongodb/v1/mongodb_provider.py

```diff
@@ -99,8 +103,12 @@ def pass_sanity_hook_checks(self) -> bool:
 
         return True
 
-    def pass_hook_checks(self, event: EventBase) -> bool:
+    def pass_hook_checks(self, event: RelationChangedEvent) -> bool:
         """Runs the pre-hooks checks for MongoDBProvider, returns True if all pass."""
+        # First, ensure that the relation is valid, useless to do anything else otherwise
+        if not self.charm.is_relation_feasible(event.relation.name):
```

Comment:
I don't think this will work for other charms that use this (namely) `mongos-k8s`, since it does not have this function.

We should either:
1. add a check here that checks if we aren't a mongos charm
OR
2. add a `is_relation_feasible ` - although I don't believe it would have any use in that charm
3.

Reply:
Oh my bad, I'll add the same check as in the other function.